src/main/webapp/src/app/nav-bar/nav-bar.component.spec.ts

```diff
@@ -0,0 +1,63 @@
+import { ComponentFixture, TestBed, inject } from '@angular/core/testing';
+import { NavBarComponent } from './nav-bar.component';
+import { Router } from '@angular/router';
+import { Location } from '@angular/common';
+import { By } from '@angular/platform-browser';
+import { CommonModule } from '@angular/common';
+import { RouterTestingModule } from '@angular/router/testing';
+import { MapComponent } from '../map/map.component';
+
+describe('NavBarComponent', () => {
+ 
+  let component: NavBarComponent;
+  let fixture: ComponentFixture<NavBarComponent>;
+  
+  beforeEach(async () => {
+    await TestBed.configureTestingModule({
+      imports: [ RouterTestingModule ],
+      declarations: [ NavBarComponent ],
+      providers: [
+        NavBarComponent
+      ]
+    })
+    .compileComponents();
+  });
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [
+        CommonModule,
+        RouterTestingModule.withRoutes([
+         { path: '', component: MapComponent }
```

Comment:
Should you include the recommendations path too and add a test for that?
it might even be better to import the routes you'll actually use from the app-routing module to make sure they don't go out of sync